File: LeetCode/Python/src/Remove_Element.py
"""
@author - Anirudh Sharma
"""
from typing import List


class RemoveElement:
    @staticmethod
    def removeElement(nums: List[int], val: int) -> int:
        # Counter for keeping track of elements other than val
        count = 0
        # Loop through all the elements of the array
        for i in range(len(nums)):
            if nums[i] != val:
                # If the element is not val
                nums[count] = nums[i]
                count += 1
        return count


if __name__ == '__main__':
    r = RemoveElement()
    print(r.removeElement([2, 3, 3, 2], 3))
    print(r.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))

File: LeetCode/JavaScript/src/remove_element.js
var removeElement = function (nums, val) {
    // Counter for keeping track of elements other than val
    let count = 0;
    // Loop through all the elements of the array
    for (let i = 0; i < nums.length; i++) {
        // If the element is not val
        if (nums[i] !== val) {
            nums[count++] = nums[i];
        }
    }
    return count;
};

console.log(removeElement([2,3,3,2], 3));
console.log(removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2));